spec/datadog/core/remote/negotiation_spec.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'datadog/core/remote/component'
+
+RSpec.describe Datadog::Core::Remote::Negotiation do
+  shared_context 'HTTP connection stub' do
+    before do
+      request_class = ::Net::HTTP::Get
+      http_request = instance_double(request_class)
+      allow(request_class).to receive(:new).and_return(http_request)
+
+      http_connection = instance_double(::Net::HTTP)
+      allow(::Net::HTTP).to receive(:new).and_return(http_connection)
+
+      allow(http_connection).to receive(:open_timeout=)
+      allow(http_connection).to receive(:read_timeout=)
+      allow(http_connection).to receive(:use_ssl=)
+
+      allow(http_connection).to receive(:start).and_yield(http_connection)
+
+      if respond_to?(:request_exception)
+        allow(http_connection).to receive(:request).with(http_request).and_raise(request_exception)
+      else
+        http_response = instance_double(::Net::HTTPResponse, body: response_body, code: response_code)
+        allow(http_connection).to receive(:request).with(http_request).and_return(http_response)
+      end
+    end
+  end
+
+  let(:settings) { Datadog::Core::Configuration::Settings.new }
+  let(:agent_settings) { Datadog::Core::Configuration::AgentSettingsResolver.call(settings, logger: nil) }
+
+  describe '#endpoint?' do
+    include_context 'HTTP connection stub'
+
+    subject(:negotiation) { described_class.new(settings, agent_settings) }
+
+    context 'when /info exists' do
+      let(:response_code) { 200 }
+      let(:response_body) do
+        {
+          'endpoints' => [
+            '/info',
+            '/foo',
+          ],
+        }.to_json
+      end
+
+      it do
+        expect(Datadog.logger).to_not receive(:error)
+
+        expect(negotiation.endpoint?('/foo')).to be true
+      end
+
+      it do
+        expect(Datadog.logger).to receive(:error).and_return(nil)
+
+        expect(negotiation.endpoint?('/bar')).to be false
+      end
+    end
+
+    context 'when /info does not exist' do
+      let(:response_code) { 404 }
+      let(:response_body) do
+        {
+          'endpoints' => [
+            '/info',
```

Comment:
If the response code is 404 probably, the body should be empty 😄 

It avoids confusion for people when reading the spec

Reply:
Just tried, but there's an issue with the way transport is done, which makes it try to parse the body nonetheless.

Reply:
Fixed